labstep/__init__.py

```diff
@@ -1,581 +1,646 @@
+
+    Returns
+    -------
+    file
+        An object representing a file on Labstep.
+    ''' 
+    headers = {'apikey': user['api_key']}
+    files = {'file': open(filepath, 'rb')}
+    url = 'https://api.labstep.com/api/generic/file/upload'
+    r = requests.post(url, headers=headers, files=files)  
+    return json.loads(r.content)
+
+def captionUploadingFile(user,entity,filepath,caption):
+    '''
+    Upload a file to a Labstep entity such as an Experiment
+    or Resource, and caption the uploading file.
+
+    Parameters
+    ----------
+    user (obj)
+        The Labstep user to attach as. Must have property 'api_key'. See 'login'.
+    entity (obj)
+        The Labstep entity to attach to. Must have 'thread' property with property 'id'.
+    filepath (str)
+        The filepath to the file to attach.
+    caption (str)
+        A caption for the file.
+
+    Returns
+    -------
+    caption
+        An object representing a comment on labstep.
+    '''
+    lsFile = uploadFile(user,filepath)
+    caption = addComment(user,caption,lsFile)  
+    return caption
+
+def editCaption(user,caption_id,caption):
+    '''
+    Edit an existing caption.
+  
+    Parameters
+    ----------
+    user (obj)
+        The labstep user. Must have property 'api_key'. See 'login'.
+    caption_id (obj)
+        The id of the caption to edit.
+    caption (str)
+        The body of the new caption.
+
+    Returns
+    -------
+    comment
+        An object representing the editted comment.
+    '''
+    headers = {'apikey': user['api_key']}
+    data = {'body': caption}
+    url = 'https://api.labstep.com/api/generic/comment/'+str(caption_id)
+    r = requests.put(url, json=data, headers=headers)
+    return json.loads(r.content)
+
+def editTag(user,tag,name):
+    '''
+    Edit the name of an existing tag.
+  
+    Parameters
+    ----------
+    user (obj)
+        The Labstep user. Must have property 'api_key'. See 'login'. 
+    tag (obj)
+        The tag to edit.
+    name (str)
+        The new name of the tag.
+
+    Returns
+    -------
+    tag
+        An object representing the editted tag.
+    '''
+    data = {'name':name}
+    headers = {'apikey': user['api_key']} 
+    url = 'https://api.labstep.com/api/generic/tag/'+ str(tag['id']) #url = url_join(API_ROOT,'/api/generic/tag/',str(tag['id']))
+    r = requests.put(url, json=data, headers=headers)
+    return json.loads(r.content)
+
+def deleteTag(user,tag):
+    '''
+    Delete an existing tag.
+  
+    Parameters
+    ----------
+    user (obj)
+        The labstep user. Must have property 'api_key'. See 'login'. 
+    tag (obj)
+        The tag to delete.
+
+    Returns
+    -------
+    tag
+        An object representing the tag to delete.
+    '''
+    headers = {'apikey': user['api_key']}
+    url = 'https://api.labstep.com/api/generic/tag/'+ str(tag['id'])
+    r = requests.delete(url, headers=headers)
+    return json.loads(r.content)
+
+def deleteExperiment(user, experiment):
```

Comment:
For `deleteExperiment` this should just call `editExperiment` with an option `{ 'deleted_at': timestamp}` as it uses the same PUT /experiment-workflow endpoint. We'll need to set up `editExperiment` so that it can take an object with extra parameters. We can go over this.